packages/components/src/navigation/item/index.tsx

```diff
@@ -79,6 +79,8 @@ export function NavigationItem( props: NavigationItemProps ) {
 		? restProps
 		: {
 				as: Button,
+				__next40pxDefaultSize:
+					'as' in restProps ? restProps.as === undefined : true,
```

Comment:
Makes sense, we want to apply `__next40pxDefaultSize` only for `Button`.

To be extra safe, should we also set it to `true` when `restProps.as` is `Button`? (just in case a consumer sets it explicitly, not sure why)

Reply:
Good edge case thinking 😄

The reasoning here is that, if there's a Button size violation the console warning will say "wp.components.Button", even if it came from within the NavigationItem black box. That can be confusing to a consumer because for all they know, they just used a `NavigationItem` and may not even have a single Button instance in their code. But as long as they did an explicit `as={ Button }`, that Button instance comes from their own code and not from within the NavigationItem black box.